src/spec.c

```diff
@@ -1501,6 +1503,19 @@ IndexSpec *NewIndexSpec(const char *name) {
   sp->used_dialects = 0;
 
   memset(&sp->stats, 0, sizeof(sp->stats));
+  
+  int res = 0;
+  pthread_rwlockattr_t attr;
+	res = pthread_rwlockattr_init(&attr) ;
+  RedisModule_Assert(res == 0);
+#if !defined(__APPLE__) && !defined(__FreeBSD__)
+	int pref = PTHREAD_RWLOCK_PREFER_WRITER_NONRECURSIVE_NP ;
```

Comment:
What happens on APPLE or FreeBSD - can Writers be starved?

Reply:
I took this one from RedisGraph. We need to double-check this, but the answer is yes